### PR TITLE
Feat/view event details

### PR DESCRIPTION
**Pull request by draskonovakovic**
This PR includes the following:

View Event Details: Display all information about an event, as well as basic details about the user who created it.
Clicking on an event opens a page showing these details.

@milan-at-2c @ivan-at-2c

### CODE REVIEW

File: src/lib/user.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import api from "./interceptor";
+
+export const getUserById = async (id: number) => {
+    try {
+      const response = await api.get("/users/" + id);
+      console.log(response.data); 
+      return response.data.data;
```
Comment (igor-at-2c) [inline src/lib/user.ts:9]: Sta ako je response undefined?

File: src/lib/event.ts [hunk 2]
```diff
@@ -17,4 +17,14 @@ export const getAllEvents = async () => {
     } catch (error: any) {
       throw new Error(error.response?.data?.message || "An error occurred while getting events.");
     }
+};
+
+export const getEventById = async (id: number) => {
+  try {
+    const response = await api.get("/events/" + id);
+    console.log(response.data); 
+    return response.data.data;
```
Comment (igor-at-2c) [inline src/lib/event.ts]: Sta ako je response undefined?

File: src/app/event-details/[id]/page.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+import { getEventById } from '@/lib/event'; 
+import { getUserById } from '@/lib/user'; 
+import { STORAGE_KEYS } from '@/constants/storageKeys';
+import { getAuthToken } from '@/utilis/authHelpers';
+
+function EventDetails() {
+  const { id } = useParams(); 
+  const router = useRouter();
+  const [isAuthenticated, setIsAuthenticated] = useState(false); 
+  const [event, setEvent] = useState<any>(null);
```
Comment (igor-at-2c) [inline src/app/event-details/[id]/page.tsx]: Zasto any?

File: src/app/event-details/[id]/page.tsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+  const [user, setUser] = useState<any>(null);
```
Comment (igor-at-2c) [inline src/app/event-details/[id]/page.tsx]: Zasto any?

File: src/app/event-details/[id]/page.tsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+import { getAuthToken } from '@/utilis/authHelpers';
+
+function EventDetails() {
+  const { id } = useParams(); 
+  const router = useRouter();
+  const [isAuthenticated, setIsAuthenticated] = useState(false); 
+  const [event, setEvent] = useState<any>(null);
+  const [user, setUser] = useState<any>(null); 
+  const [loading, setLoading] = useState(true); 
+  const [error, setError] = useState<string | null>(null);
```
Comment (igor-at-2c) [inline src/app/event-details/[id]/page.tsx]: Zasto null a ne prazan string?

File: src/app/event-details/[id]/page.tsx [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+  const formatDate = (dateString: string) => {
+    try {
+      const date = new Date(dateString);
```
Comment (igor-at-2c) [inline src/app/event-details/[id]/page.tsx:26]: Provera da li je data falsy?

**Review by igor-at-2c — CHANGES_REQUESTED** [review]

File: src/app/event-details/[id]/page.tsx (hunk 3 above)
Comment (draskonovakovic) [reply src/app/event-details/[id]/page.tsx]: Zato sto sam budala, zaboravio sam da sam Event tip napravio. Sredicu

File: src/app/event-details/[id]/page.tsx [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+  const fetchEventData = async (eventId: string | string[]) => {
+    try {
+      const eventData = await getEventById(+eventId);
+      if (!eventData) {
+        throw new Error("Event not found");
+      }
+      setEvent(eventData);
+
+      if (eventData.creator_id) {
```
Comment (igor-at-2c) [inline src/app/event-details/[id]/page.tsx]: Bolje uradi early return ako ne postoji creator_id

File: src/app/event-details/[id]/page.tsx [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    const token = getAuthToken();
+    setIsAuthenticated(!!token);
+
+    if (id) {
```
Comment (igor-at-2c) [inline src/app/event-details/[id]/page.tsx]: Bolje ti je da uradis early return ako ne postoji id

File: src/lib/event.ts [hunk 9]
```diff
@@ -1,20 +1,39 @@
...
 export const getAllEvents = async () => {
     try {
-      const response = await api.get("/events/");
-      console.log(response.data); 
-      return response.data.data;
+        const response = await api.get("/events/");
+        if (!response || !response.data) {
+            throw new Error("Unexpected response: Response or data is undefined.");
+        }
+        console.log(response.data);
```
Comment (igor-at-2c) [inline src/lib/event.ts]: Gledaj da brises console.log, nisu ti potrebni na frontu osim ako i dalje testiras neki feature

**Review by igor-at-2c — CHANGES_REQUESTED** [review]

**Review by igor-at-2c — APPROVED** [review]

**Pull request merged**